plugins/CoreAdminHome/CoreAdminHome.php

```diff
@@ -51,6 +52,8 @@ public function addSystemSummaryItems(&$systemSummary)
     public function cleanupUser($userLogin)
     {
         PluginSettingsTable::removeAllUserSettingsForUser($userLogin);
+
+        SegmentEditor::transferAllUserSegmentsToSuperUser($userLogin);
```

Comment:
btw usually we would call this method in `Piwik\Plugins\SegmentEditor\SegmentEditor` class by listening to the `UsersManager.deleteUser` event see https://developer.matomo.org/guides/events . There we would then call the method to delete all segments for the deleted user. Basically the same way it's done in this class. It's not needed to do it that way but usually best practice to contain all the logic that belongs to the plugin in the actual plugin.

Reply:
I like that, more clean solution. I will modify it.